File: data/java/PROGRAM_TO_CHECK_IF_A_GIVEN_NUMBER_IS_LUCKY_ALL_DIGITS_ARE_DIFFERENT_TRANSLATED.py
====================
def f_gold ( n ) :
    arr = [ ]
    for i in range ( 10 ) :
        arr.append ( False )
    while n > 0 :
        digit = n % 10
        if arr [ digit ] :
            return False
        arr.append ( True )
        n = n / 10
    return True

File: data/python/PROGRAM_TO_CHECK_IF_A_GIVEN_NUMBER_IS_LUCKY_ALL_DIGITS_ARE_DIFFERENT_TRANSLATED.java
====================
public static final int f_gold ( int n ) {
  int [ ] ar = new int [ 10 ] ;
  while ( ( n > 0 ) && ( ar [ n ] != 0 ) ) {
    int digit = Math . floor ( n % 10 ) ;
    if ( ( ar [ digit ] != 0 ) && ( ar [ digit ] != 1 ) ) {
      return 0 ;
    }
    ar [ digit ] = 1 ;
    n = n / 10 ;
  }
  return 1 ;
}

